src/lwc/geFormWidgetTokenizeCard/geFormWidgetTokenizeCard.js

```diff
@@ -62,36 +61,56 @@ export default class geFormWidgetTokenizeCard extends LightningElement {
 
     set widgetDataFromState(widgetState) {
         this._widgetDataFromState = widgetState;
-        this.handleWidgetDataChange(widgetState);
+
+        if (isEmptyObject(this.PAYMENT_TRANSACTION_STATUS_ENUM) ||
+            this.shouldHandleWidgetDataChange()) {
+
+            this.handleWidgetDataChange();
+        }
     }
 
-    handleWidgetDataChange(widgetState) {
+    handleWidgetDataChange() {
         this._hasPaymentMethodInTemplate =
             this.sourceFieldsUsedInTemplate.includes(apiNameFor(DATA_IMPORT_PAYMENT_METHOD));
 
         if (this._hasPaymentMethodInTemplate) {
-            this._currentPaymentMethod = widgetState[apiNameFor(DATA_IMPORT_PAYMENT_METHOD)];
+            this._currentPaymentMethod = this.widgetDataFromState[apiNameFor(DATA_IMPORT_PAYMENT_METHOD)];
 
             if (this.hasValidPaymentMethod(this._currentPaymentMethod)) {
                 if (this.isMounted) {
                     this.requestSetPaymentMethod(this._currentPaymentMethod);
                 } else {
-                    this.handleUserEnabledWidget();
+                    if (!this.hasUserDisabledWidget) {
+                        this.handleUserEnabledWidget();
+                    }
                 }
             } else {
-                this.handleUserDisabledWidget();
+                this.toggleWidget(true, this.disabledWidgetMessage);
```

Comment:
I think this change in commit d03838056 the culprit for the unit test that is currently failing, `c-ge-form-widget-tokenize-card › should render extended disabled message without button to re-enable widget`. Do we still need to set `this.hasUserDisabledWidget` here when disabling the widget?

Reply:
@bdvorachek we had to change this to resolve a bug that calling this function introduced which currently exists in the 230 branch.